mempool/reactor.go

```diff
@@ -134,16 +160,40 @@ func (memR *MempoolReactor) broadcastTxRoutine(peer Peer) {
 				continue
 			}
 		}
-		// send memTx
-		msg := &TxMessage{Tx: memTx.tx}
+
+		// Fetch txs to send.
+		// If there's only one, we send the TxMessage (for backwards compat).
+		// Otherwise, use TxsMessage to send many txs.
+		last := next
+		nTxs := 0
+		for i := 0; i < batchSize; i++ {
+			nTxs += 1
+			memTxs[i] = memTx.tx
+			last = next
+			next = last.Next()
+
+			if next == nil {
+				break
+			}
+			memTx = next.Value.(*mempoolTx)
+		}
+
+		var msg MempoolMessage
+		if nTxs > 1 {
+			msg = &TxsMessage{Txs: memTxs[:nTxs]}
+		} else {
+			msg = &TxMessage{Tx: memTxs[0]}
+		}
+
 		success := peer.Send(MempoolChannel, struct{ MempoolMessage }{msg})
```

Comment:
what about backword-compatibility? will old versions crash upon receiving MempoolMessage with `Txs`?

Reply:
They shouldn't. See https://github.com/tendermint/tendermint/blob/master/mempool/reactor.go#L74.

We could probably do with a test for this. 

That said according to Jae's comments this might be abandoned if we can figure out the performance issue ...

Reply:
I see. Thanks